src/main/java/org/folio/circulation/resources/RequestNoticeSender.java

```diff
@@ -306,29 +319,35 @@ private CompletableFuture<Result<Void>> sendNoticeOnRecall(Request request) {
       return ofAsync(null);
     }
 
-    PatronNoticeEvent itemRecalledEvent = new PatronNoticeEventBuilder()
-      .withItem(loan.getItem())
-      .withUser(loan.getUser())
-      .withEventType(ITEM_RECALLED)
-      .withNoticeContext(createLoanNoticeContext(loan))
-      .withNoticeLogContext(NoticeLogContext.from(loan))
-      .build();
-
-    eventPublisher.publishRecallRequestedEvent(loan);
-
-    return patronNoticeService.acceptNoticeEvent(itemRecalledEvent);
+    return fetchAdditionalInfo(request)
+      .thenAccept(r -> r.map(req -> {
+        Loan updatedLoan = req.getLoan();
+        PatronNoticeEvent itemRecalledEvent = new PatronNoticeEventBuilder()
+          .withItem(updatedLoan.getItem())
+          .withUser(updatedLoan.getUser())
+          .withEventType(ITEM_RECALLED)
+          .withNoticeContext(createLoanNoticeContext(updatedLoan))
+          .withNoticeLogContext(NoticeLogContext.from(updatedLoan))
+          .build();
+
+        eventPublisher.publishRecallRequestedEvent(updatedLoan);
+
+        return patronNoticeService.acceptNoticeEvent(itemRecalledEvent);
+      }))
+      .thenApply(r -> null);
```

Comment:
Please extract `req -> {...}` into a method.
```suggestion
    return fetchAdditionalInfo(request)
      .thenCompose(r -> r.after(req -> {
        Loan updatedLoan = req.getLoan();
        PatronNoticeEvent itemRecalledEvent = new PatronNoticeEventBuilder()
          .withItem(updatedLoan.getItem())
          .withUser(updatedLoan.getUser())
          .withEventType(ITEM_RECALLED)
          .withNoticeContext(createLoanNoticeContext(updatedLoan))
          .withNoticeLogContext(NoticeLogContext.from(updatedLoan))
          .build();

        eventPublisher.publishRecallRequestedEvent(updatedLoan);

        return patronNoticeService.acceptNoticeEvent(itemRecalledEvent);
      }));
```